perfkitbenchmarker/pkb.py

```diff
@@ -29,25 +29,29 @@
          will then return the machine to the state it was at before Prepare
          was called.
 
-PerfKitBenchmarker has following run stages: prepare, run, cleanup and all.
-
-prepare: PerfKitBenchmarker will read command-line flags, decide which
-         benchmarks to run and create necessary resources for each benchmark,
-         including networks, VMs, disks, keys and execute the Prepare function
-         of each benchmark to install necessary software, upload datafiles, etc
-         and generate a run_uri, which can be used to run benchmark multiple
-         times.
+PerfKitBenchmarker has the following run stages: provision, prepare,
+    run, cleanup, teardown, and all.
+
+provision: Read command-line flags, decide what benchmarks to run, and
+    create the necessary resources for each benchmark, including
+    networks, VMs, disks, and keys, and generate a run_uri, which can
+    be used to run the benchmark multiple times.
```

Comment:
For the 'provision' stage, I'd say the run_uri is used to resume execution at later stages of the same run. The fact that the 'run' stage can be repeated is covered in the description of the 'run' stage below.